legend-engine-xt-serviceStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/service/auth/ServiceStoreConnectionProvider.java

```diff
@@ -0,0 +1,194 @@
+public class ServiceStoreConnectionProvider extends ConnectionProvider<HttpURLConnection>
+{
+    private static AuthenticationMethodProvider authenticationMethodProvider;
+
+    public ServiceStoreConnectionProvider(AuthenticationMethodProvider authenticationMethodProvider) {
+        this.authenticationMethodProvider = authenticationMethodProvider;
+    }
+
+    public HttpURLConnection makeConnection(ConnectionSpec connectionSpec, AuthenticationSpec authenticationSpec, Identity identity) throws Exception
+    {
+        assert (connectionSpec instanceof ServiceStoreConnectionSpec);
+        ServiceStoreConnectionSpec serviceStoreConnectionSpec = (ServiceStoreConnectionSpec) connectionSpec;
+
+        assert(authenticationSpec instanceof ServiceStoreAuthenticationSpec);
+        ServiceStoreAuthenticationSpec serviceStoreAuthenticationSpec = (ServiceStoreAuthenticationSpec) authenticationSpec;
+
+        Map<String, SecurityScheme> securitySchemeMap = serviceStoreAuthenticationSpec.securitySchemes;
+        Map<String, AuthenticationSpec> authenticationSpecMap = serviceStoreAuthenticationSpec.authSpecs;
+
+        for (Map.Entry<String, SecurityScheme> entry : securitySchemeMap.entrySet())
+        {
+            Credential cred = processSecurityScheme(entry.getKey(),entry.getValue(),authenticationSpecMap,identity);
+            HttpURLConnection conn = makeConnectionUtil(serviceStoreConnectionSpec,serviceStoreAuthenticationSpec,identity);
+            configureAuthentication(conn,entry.getKey(),entry.getValue(),cred);
+            if (conn.getResponseCode() == HttpStatus.SC_OK)
+            {
+                return conn;
+            }
+        }
+        return makeConnectionUtil(serviceStoreConnectionSpec,serviceStoreAuthenticationSpec,identity);
+    }
+
+    public HttpURLConnection makeConnectionUtil(ServiceStoreConnectionSpec serviceStoreConnectionSpec, AuthenticationSpec authenticationSpec, Identity identity) throws Exception
+    {
+        HttpURLConnection connection = (HttpURLConnection) (new URL(serviceStoreConnectionSpec.uri.toString()).openConnection());
+        switch (serviceStoreConnectionSpec.httpMethod.toString())
+        {
+            case "GET":
+                connection.setRequestMethod("GET");
+                break;
+            case "POST":
+                connection.setRequestMethod("POST");
+                break;
+            default:
+                throw new UnsupportedOperationException("The HTTP method " + serviceStoreConnectionSpec.httpMethod + " is not supported");
+        }
+        serviceStoreConnectionSpec.headers.forEach(header -> connection.setRequestProperty(header.getName(),header.getValue()));
+
+        return connection;
+    }
+
+    private static Credential processSecurityScheme(String schemeId, SecurityScheme securityScheme, Map<String,AuthenticationSpec> authenticationSpecMap, Identity identity) throws Exception
```

Comment:
maybe if we have composite auth spec we won't have this here but ata generic sapce